### PR TITLE
fix: nil pointer dereference

### PR DESCRIPTION
**Pull request by tao12345666333**
Signed-off-by: Jintao Zhang <zhangjintao9020@gmail.com>

Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

___
### Bugfix

   This was introduced through #516  while resolving the conflict. :crying_cat_face:

### CODE REVIEW

**Review by tokers — APPROVED** [review]

**Comment by tao12345666333** [comment]
ApisixConsumer's e2e test failed because it triggered the max length limit of consumer name in APISIX. Current limit is 32 
https://github.com/apache/apisix/blob/10677e9067b43fb3b2555c917c322b4454b22c77/apisix/schema_def.lua#L623

We have two solutions

1. Increase the limit on the length of APISIX consumer name. https://github.com/apache/apisix/blob/10677e9067b43fb3b2555c917c322b4454b22c77/apisix/schema_def.lua#L623

2. Shorten the namespace name generated by the current e2e test https://github.com/apache/apisix-ingress-controller/blob/2bf4b6be29648c1c3e98006edae50252a0555a08/test/e2e/scaffold/scaffold.go#L262

I tried to create ApisixConsumer in other namespaces, but ApisixIngress currently only monitors the namespace where it is located. A third solution can be derived here, which is to update deployment in this testcase.

But I think the second method has the lowest cost.

cd @tokers @gxthrj  @membphis

**Comment by tao12345666333** [comment]
I have opened a PR to change APISIX consumer name's max length. https://github.com/apache/apisix/pull/4379

**Review by gxthrj — APPROVED** [review]

**Comment by tokers** [comment]
@tao12345666333 So we have to wait until the APISIX PR merged?

**Comment by tao12345666333** [comment]
@tokers Both methods I mentioned above are fine. 

https://github.com/apache/apisix/pull/4379 should be merged soon.

**Comment by gxthrj** [comment]
@tao12345666333 some tests failed.

**Comment by tokers** [comment]
@tao12345666333 I think option 2 is a quick way to let this PR to be merged.

**Comment by tao12345666333** [comment]
https://github.com/apache/apisix/pull/4379 has been merged. It won't block us anymore.

Now there is only one failed test case, which seems to be irrelevant to the change.
https://github.com/apache/apisix-ingress-controller/pull/529/checks?check_run_id=2762431556#step:6:2154

@tokers @gxthrj PTAL.

File: pkg/types/apisix/v1/types.go
Comment (tao12345666333) [inline pkg/types/apisix/v1/types.go:460]: should we change APISIX consumer name's schema?

Comment (tao12345666333) [reply pkg/types/apisix/v1/types.go:460]: change to lower priority.
Priority to maintain compatibility with APISIX v2.6

**Pull request merged**
